app/constants/sso.ts

```diff
@@ -7,15 +7,43 @@ export const REDIRECT_URL_SCHEME = 'mmauth://';
 export const REDIRECT_URL_SCHEME_DEV = 'mmauthbeta://';
 
 const constants = keyMirror({
+    SAML: null,
     GITLAB: null,
     GOOGLE: null,
     OFFICE365: null,
     OPENID: null,
-    SAML: null,
 });
 
+const values = {
+    SAML: {
+        defaultMessage: 'SAML',
+        compassIcon: 'lock',
+        id: 'mobile.login_options.saml',
+    },
+    GITLAB: {
+        defaultMessage: 'GitLab',
+        imageSrc: require('@assets/images/Icon_Gitlab.png'),
+        id: 'mobile.login_options.gitlab',
+    },
+    GOOGLE: {
+        defaultMessage: 'Google',
+        imageSrc: require('@assets/images/Icon_Google.png'),
+        id: 'mobile.login_options.google',
+    },
+    OFFICE365: {
+        defaultMessage: 'Office 365',
+        imageSrc: require('@assets/images/Icon_Office.png'),
+        id: 'mobile.login_options.office365',
+    },
+    OPENID: {
+        defaultMessage: 'Open ID',
+        id: 'mobile.login_options.openid',
+    },
+};
```

Comment:
this should no be included in this constants file.. why? cause the constants are initialized and we require the images even if the user won't ever use SSO or even after the user has logged in, thus loading assets and kept in memory without need.